### PR TITLE
chore(upgrade): upgrade carbon deps to v10.22.0

### LINKED ISSUE #4200: React: Masthead - With tab/shift tab key, focus is not going to the profile elements.
## Detailed description

Issue: 
With tab/shift tab key, focus is not going to the "My IBM, Log in" elements.

Expected :
With tab/shift tab key, Focus should go to each and every interactive element and it should be operable thru keyboard.

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/4272, https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/4200

### Description

Upgrade Carbon dependencies to latest `v10.22.0`. This also fixes #4200 which was caused by Carbon `v10.21.0`.

### Changelog

**Changed**

- Carbon dependencies upgraded to `v10.22.0`

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
Thank you for taking this on @kennylam! Could you update the snapshot and check the diff? Thanks!

**Comment by kennylam** [comment]
> Thank you for taking this on @kennylam! Could you update the snapshot and check the diff? Thanks!

Doh! Good catch thanks @asudoh!

File: packages/styles/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "@carbon/import-once": "10.3.0",
     "@carbon/layout": "10.13.0",
     "@carbon/motion": "10.9.0",
-    "@carbon/themes": "10.20.0",
+    "@carbon/themes": "10.21.0",
     "@carbon/type": "10.16.0",
     "carbon-components": "10.21.0"
```
Comment (jeffchew) [inline packages/styles/package.json:43]: @kennylam it doesn't look like all components were updated, did you run the script `yarn upgrade-carbon` to get all of the carbon dependencies?

Comment (kennylam) [reply packages/styles/package.json:43]: @jeffchew I did. not sure why that one didn't get picked up

**Pull request merged**